neo/SmartContract/InteropService.cs

```diff
@@ -336,144 +307,73 @@ private static bool Blockchain_GetBlock(ApplicationEngine engine)
                 hash = new UInt256(data);
             else
                 return false;
-            if (hash == null)
-            {
-                engine.CurrentContext.EvaluationStack.Push(new byte[0]);
-            }
+
+            Block block = hash != null ? engine.Snapshot.GetBlock(hash) : null;
+            if (block == null)
+                engine.CurrentContext.EvaluationStack.Push(StackItem.Null);
             else
-            {
-                Block block = engine.Snapshot.GetBlock(hash);
-                engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(block));
-            }
+                engine.CurrentContext.EvaluationStack.Push(block.ToStackItem());
             return true;
         }
 
         private static bool Blockchain_GetTransaction(ApplicationEngine engine)
         {
             byte[] hash = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
             Transaction tx = engine.Snapshot.GetTransaction(new UInt256(hash));
-            engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(tx));
+            if (tx == null)
+                engine.CurrentContext.EvaluationStack.Push(StackItem.Null);
+            else
+                engine.CurrentContext.EvaluationStack.Push(tx.ToStackItem());
             return true;
         }
 
         private static bool Blockchain_GetTransactionHeight(ApplicationEngine engine)
         {
             byte[] hash = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
-            int? height = (int?)engine.Snapshot.Transactions.TryGet(new UInt256(hash))?.BlockIndex;
-            engine.CurrentContext.EvaluationStack.Push(height ?? -1);
+            var tx = engine.Snapshot.Transactions.TryGet(new UInt256(hash));
+            engine.CurrentContext.EvaluationStack.Push(tx != null ? new BigInteger(tx.BlockIndex) : BigInteger.MinusOne);
             return true;
         }
 
-        private static bool Blockchain_GetContract(ApplicationEngine engine)
+        private static bool Blockchain_GetTransactionFromBlock(ApplicationEngine engine)
         {
-            UInt160 hash = new UInt160(engine.CurrentContext.EvaluationStack.Pop().GetByteArray());
-            ContractState contract = engine.Snapshot.Contracts.TryGet(hash);
-            if (contract == null)
-                engine.CurrentContext.EvaluationStack.Push(new byte[0]);
+            byte[] data = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            UInt256 hash;
+            if (data.Length <= 5)
+                hash = Blockchain.Singleton.GetBlockHash((uint)new BigInteger(data));
```

Comment:
Perhaps here we should decide if format is `uint32` or `BigInteger`... they are different and may cause problems.
Example: some positive numbers may require `00` prefix (`le` suffix) to avoid being negative... in typical `uint` this doesnt happen, since it's unsigned anyway. If we want to enforce `uint` here, perhaps it's better to change `GetBlockHash` method to accept BigInteger as input, and also serialize `height` as biginteger.

Reply:
To be honest, I think this may be a problem... I'll try to fix this during break times in conference today.

Reply:
There is no problem. `BigInteger` is a superset of `uint`.